kvstore/store.go

```diff
@@ -0,0 +1,41 @@
+package kvstore
+
+import (
+	"errors"
+	"github.com/syndtr/goleveldb/leveldb/iterator"
+)
+
+// Store implements a key-value store. May be backed by a database like LevelDB.
+type Store interface {
```

Comment:
I'm not quite sure if this will be the right level to abstract. It may be a can of worms, e.g. if you want to specify the options in these APIs, then you need to keep adding more structs to hide different DBs.
It's possible the abstraction at application level will be more appropriate. These are very thin wrappers anyway.

Reply:
I've now moved it to `eigenda/common/kvstore` (from your request to unify the interfaces).